lib/ex_format/state.ex

```diff
@@ -0,0 +1,29 @@
+defmodule ExFormat.State do
```

Comment:
Awesome, didn't know about this! Btw what do you mean by compile-time guarantees in this context?